client/src/components/NavBar.tsx

```diff
@@ -1,17 +1,26 @@
+import { ReactComponent as Logo } from 'assets/MHPLogo.svg';
 import React from 'react';
-import { NavLink } from 'react-router-dom';
 import { Nav, Navbar, NavDropdown } from 'react-bootstrap';
-import { useBikeVersion, bikeVersions } from 'router';
-import { ReactComponent as Logo } from 'assets/MHPLogo.svg';
+import { NavLink } from 'react-router-dom';
+import { VersionInfo } from '../router';
 import styles from './NavBar.module.css';
 
+/**
+ * @property {VersionInfo} bikeVersion Version info for bike
+ * @property {VersionInfo[]} bikeVersions Version info for all bikes
+ */
+export interface NavBarProps {
+  bikeVersion: VersionInfo | null,
+  bikeVersions: VersionInfo[]
+}
+
 /**
  * Navigation bar component
  *
- * @returns {React.Component} Component
+ * @param {NavBarProps} props Props
+ * @returns {JSX.Element} Component
  */
-export default function NavBar() {
-  const bikeVersion = useBikeVersion();
+export default function NavBar({ bikeVersion, bikeVersions }: NavBarProps): JSX.Element {
```

Comment:
Is there a reason the params are passed as a single props object rather than separately?

Reply:
Isn't that just how functional components work? Their parameters are all passed in as a single object which can then be destructured.

Reply:
Yeah you're right actually, not sure what I was thinking. (I've never really been sure why it's done like this though, would be interested to know)

Reply:
Probably just makes it easier for developers so we don't have to remember the position of the parameters. E.g. if you have `Component(arg1, arg2)` then you would have to do `<Component arg1={arg1} arg2={arg2}>`. Also having an object makes it easier to skip parameters.